### PR TITLE
docs: add description + code example to `structuredClone`, `CompressionStream`, `DecompressionStream`

### PR DESCRIPTION
**Pull request by GJZwiers**
Also this updates some more jsdoc that is not rendering well on hover in VS Code.

The compression examples are from the 1.19 release notes, thought it would be nice to add them in

### CODE REVIEW

File: ext/web/lib.deno_web.d.ts [hunk 1]
```diff
@@ -805,19 +809,73 @@ declare class MessagePort extends EventTarget {
   ): void;
 }
 
+/**
+ * Creates a deep copy of a given value using the structured clone algorithm.
+ *
+ * Unlike a shallow copy, a deep copy does not hold the same references as the source object, meaning its properties can be changed without affecting the source.
+ * For more details, see [MDN](https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy).
```
Comment (bartlomieju) [inline ext/web/lib.deno_web.d.ts:816]: Could you wrap at 80 characters?

File: cli/dts/lib.deno.window.d.ts [hunk 2]
```diff
@@ -83,9 +85,11 @@ declare function addEventListener(
 
 /** Remove a previously registered event listener from the global scope
  *
- *     const lstnr = () => { console.log('hello'); };
- *     addEventListener('load', lstnr);
- *     removeEventListener('load', lstnr);
+ * ```ts
+ * const lstnr = () => { console.log('hello'); };
+ * addEventListener('load', lstnr);
+ * removeEventListener('load', lstnr);
```
Comment (bartlomieju) [inline cli/dts/lib.deno.window.d.ts:91]:
```suggestion
 * const listener = () => { console.log('hello'); };
 * addEventListener('load', listener);
 * removeEventListener('load', listener);
```

File: ext/web/lib.deno_web.d.ts (hunk 1 above)
Comment (GJZwiers) [reply ext/web/lib.deno_web.d.ts]: Done!

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @GJZwiers!

**Pull request merged**